fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -558,6 +558,26 @@ private void addResourceLinkHeaders(final FedoraResource resource, final boolean
                 servletResponse.addHeader(LINK, buildLink(VERSIONING_TIMEMAP_TYPE, "type"));
             }
         }
+
+        // Add user-provided types as Link headers... when a description exists
+        if (resource.getDescription() != null) {
+            for (final URI typeURI : resource.getDescription().getTypes()) {
+
+                // Get namespace of type
+                final String type = typeURI.toString();
+                final String namespace;
+                if (type.contains("#")) {
+                    namespace = type.substring(0, type.indexOf('#') + 1);
+                } else {
+                    namespace = type.substring(0, type.lastIndexOf('/') + 1);
+                }
+
+                // Omit server-managed types
+                if (!isManagedNamespace.test(namespace)) {
```

Comment:
The intent of this is unclear.  Is it to avoid duplicate link headers due to the fact that some managed namespaces are added elsewhere in the code (e.g. LDP types, memento types), or is the intent to assure that the types in the `fedora` ontology are not exposed, or both?

Reply:
The intent of this check is to prevent duplicate `Link` headers, as the server-manged types are added elsewhere.

Reply:
In that case, there may be a bug, as it has the side effect of removing types in the fedora namespace, as noted in the [original review comment](https://github.com/fcrepo4/fcrepo4/pull/1501#pullrequestreview-189032473):
> for example <> rdf:type <fedora:Resource> appears in RDF, but not as a link header. 

To show this, create a binary, and look at the triples in the /fcr:metadata resource.  You'll see something like:

        rdf:type                 fedora:NonRdfSourceDescription ;
        rdf:type                 fedora:Binary ;
        rdf:type                 <http://example.org/Bar> ;
        rdf:type                 <http://example.org#Foo> ;
        rdf:type                 fedora:Resource ;

Yet in the headers, the "fedora:Resource, fedora:Binary" are not present:

    Link: <http://www.w3.org/ns/ldp#RDFSource>; rel="type"
    Link: <http://mementoweb.org/ns#OriginalResource>; rel="type"
    Link: <http://mementoweb.org/ns#TimeGate>; rel="type"
    Link: <http://example.org/Bar>; rel="type"
    Link: <http://example.org/#Foo>; rel="type"

Reply:
@birkland My impression is that the intent is to not display any of the server managed rdf types in the fedora namespace via the link headers (fedora:NonRdfSourceDescription is not present either). I don't know the reasons for that decision, except perhaps that they aren't included in the specification.

Reply:
@birkland : It does not seem like the behavior you are observing would be related to updates in this PR. Are you witnessing the presence of `fedora:Resource` and `fedora:Binary` Link headers in `master`?

Reply:
@bbpennel  @awoods I'm just trying to figure out what the desired relationship between between `rdf:type` expressed in RDF vs `Link` headers is supposed to be.  The jira ticket suggests it should be all rdf:type values:

> It would be helpful to clients if the header included ***all rdf:type values*** so that they could detect locally meaningful resource types

As demonstrated, his PR does _not_ include all rdf:type values present in the RDF.  If that is indeed the expected behaviour, then I think a clearer definition of the contract regarding exposed link headers needs to be articulated.   See also [my comment on writing this up as an RFC](https://github.com/fcrepo4/fcrepo4/pull/1501#issuecomment-451182902).  If another implementation wanted to do this, what sort of logic would they need to employ?

Reply:
@birkland : I am reluctant to encourage the use of the `fedora:NonRdfSourceDescription`,    `fedora:Binary`, `fedora:*` RDF types in cases where the same information is represented by more standard ontologies... that we already use.

It would have been good to pull those out in 5.0.0... but we can wait for 6.0.0.

Reply:
@awoods So is the contract this PR offers to clients one of the following?  

In addition to mandatory link rel=type headers required by the fedora specifications (LDP, memento, etc):
* A) link rel=type headers shall contain all rdf:types present in the body?
* B) link rel=type headers shall contain all non-server-managed rdf:types found in the body?
* C) link rel=type headers shall include all rdf:types present in the body, but we are making an exception for fcrepo5 because one of the properties are deprecated.  If Fedora5+ ever adds a new, non-deprecated, server-managed rdf:type, then we'll be bound to expose it in the headers.

Reply:
@birkland : At this point, I think option `B` is the most correct.

Reply:
@awoods OK.  Mind if I update the jira ticket with the following text:

The desired functionality is as follows.  In addition to mandatory link rel=type headers required by the fedora specifications (LDP, memento, etc):
* On LDP-RS (RDFSource) resources, Fedora shall expose link rel=type headers for all non-server-managed rdf:type triples found in its body whose subject is the LDP-RS
* On LDP-NR (NonRDFSource) resources, Fedora shall expose Link rel=type headers for all non-server-managed rdf:type triples found in the body of its descriptive LDP-RS whose subject is the LDP-NR
* On LDP-RS resources that describe an LDP-NR, Fedora shall expose Link rel=type headers for all non-server-managed rdf:type triples found in its body whose subject is either the LDP-RS, or the described LDP-NR

Reply:
@birkland : please do. Thank you!